src/Promise.fs

```diff
@@ -454,6 +454,142 @@ let Parallel (pr: seq<JS.Promise<'T>>): JS.Promise<'T[]> = jsNative
 /// <returns>Return a new promise returning an array containing all the promise result</returns>
 let all (pr: seq<JS.Promise<'T>>): JS.Promise<'T[]> = jsNative
 
+[<StringEnum>]
+type SettledStatus =
+    | Fulfilled
+    | Rejected
+
+type SettledValue<'T> = 
+    {| status: SettledStatus; value: 'T option; reason: exn option |}
+
+let inline (|Success|Rejection|) (p: SettledValue<'T>) =
+    match p.status with 
+    | Fulfilled ->  Success p.value.Value
+    | Rejected -> Rejection p.reason.Value
+
+/// <summary>
+/// Helper functionm to convert a settled result into an F# result
+/// </summary>
+/// <param name="p">The settled result</param>
+/// <typeparam name="'T"></typeparam>
+/// <returns>The F# result</returns>
+/// <example>
+/// <code lang="fsharp">
+/// Promise.allSettled [success; rejection]
+/// |> Promise.map (fun results ->
+///     let success = results.[0]
+///     let rejection = results.[1]
+///     match success |> Promise.settledResult with
+///     | Ok value -> value |> equal 1
+///     | _ -> failwith "Unreachable result"
+///     
+///     match rejection |> Promise.settledResult with
+///     | Error ex -> ex.Message |> equal "I Failed"
+///     | _ -> failwith "Unreachable result"
+/// )
+/// </code>
+/// </example>
+let settledResult (p: SettledValue<'T>) =
+    match p with
+    | Success value -> Ok value
+    | Rejection err -> Error err
```

Comment:
This gives many options to deal with settled values: 1. access the original interface, 2. use an active pattern, 3. use settledResult helper to convert it to F# Result. I regret a bit that we did something similar with the Promise.catch combinations, and it's a bit confusing now (specially because signatures are not consistent). Maybe at first I'll just offer one way: you can include some helper code to convert everything to F# Result directly.

Reply:
I'll leave the `settledResult` with direct conversion to F# `Result`, yeah I felt it could be confusing as well, I think Result is more consistent with the rest of the library so I'll follow your suggestion there